src/Messages/Attachments/TSAttachment.m

```diff
@@ -33,18 +34,34 @@ - (instancetype)initWithServerId:(UInt64)serverId
     return self;
 }
 
+- (BOOL)isDecimalNumberText:(NSString *)text {
+    return [text componentsSeparatedByCharactersInSet:[NSCharacterSet decimalDigitCharacterSet]].count == 1;
+}
+
 - (nullable instancetype)initWithCoder:(NSCoder *)coder
 {
     self = [super initWithCoder:coder];
     if (!self) {
         return self;
     }
 
-    if (_attachmentSchemaVersion < 2) {
```

Comment:
> Regardless of this fix, we should be properly setting the TSAttachmentSchemaVersion for subclasses of TSAttachment (e.g. with designated initializers in this class), else every future migration will likely suffer the same problem.

I've modified `TSAttachmentStream` to use initializers in `TSAttachment` so that `TSAttachmentSchemaVersion` is set for new instances.  I'm using these new initializers to ensure that new  `TSAttachmentStream` that represented downloaded incoming attachments no longer discard their server id and encryption key, with an eye towards supporting re-download in the future.

> And if we're going to do that, I think we could/should remove these changes.

These changes are still necessary, because`TSAttachmentSchemaVersion` still won't be set for legacy instances of `TSAttachmentStream`.  I'm going to simplify things and leverage the changes made in https://github.com/WhisperSystems/SignalServiceKit/pull/129 and move this to `TSAttachmentPointer`.